source/core/ExecutionContext.cpp

```diff
@@ -450,6 +450,24 @@ Int32 /*ExecSlicesResult*/ ExecutionContext::ExecuteSlices(Int32 numSlices, Int3
     return reply;
 }
 //------------------------------------------------------------
+void ExecutionContext::ExecuteTillNextStopPoint()
+{
+    int sizeOfQueue = _runQueue.size();
+    while (sizeOfQueue > 0)
+    {
+        VIClump*  clump = _runQueue.Dequeue();
+        InstructionCore* currentInstruction = _runningQueueElt ? _runningQueueElt->_savePc : nullptr;
+        InstructionCore* nextInstruction = nullptr;
+        while (nextInstruction != &_culDeSac) {
+            currentInstruction = nextInstruction ? nextInstruction : currentInstruction;
+            nextInstruction = _PROGMEM_PTR(currentInstruction, _function)(currentInstruction);
+        }
+        clump->_savePc = currentInstruction;
```

Comment:
Would we always have a _savePc here? What if the DebugPoint was the last instruction in that clump?